### PR TITLE
Core: Move InMemoryCatalog from test to core

### LINKED ISSUE #7115: Move InMemoryCatalog to core
### Feature Request / Improvement

Currently the related classes are in test, I think it might also be useful to move it to core, so people can import the class and use it in external tests.

Any thoughts? @nastra 

### Query engine

None

### PR DESCRIPTION
**Pull request by nastra**
fixes #7115

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/inmemory/InMemoryOutputFile.java [hunk 1]
```diff
@@ -30,7 +33,7 @@ public class InMemoryOutputFile implements OutputFile {
 
   private boolean exists = false;
   private ByteArrayOutputStream contents;
-  private InMemoryFileIO parentFileIO;
+  private final InMemoryFileIO parentFileIO;
```
Comment (jackye1995) [inline core/src/main/java/org/apache/iceberg/inmemory/InMemoryOutputFile.java:36]: nit: can we place final variables over non final ones

Comment (nastra) [reply core/src/main/java/org/apache/iceberg/inmemory/InMemoryOutputFile.java]: sure. done

**Review by jackye1995 — APPROVED** [review]
looks good to me!

**Review by amogh-jahagirdar — APPROVED** [review]

**Comment by jackye1995** [comment]
Thanks for the work @nastra, I will go ahead to merge instead of waiting for more reviews since it's already reviewed during the original PR and this is just to move the class, and we have a good reason to move it.

**Pull request merged**